platforms/defaults_windows.go

```diff
@@ -78,14 +81,21 @@ func revision(v string) int {
 	return r
 }
 
+func prefix(v string) string {
+	parts := strings.Split(v, ".")
+	if len(parts) < 4 {
+		return v
+	}
+	return strings.Join(parts[0:2], ".")
+}
+
 // Default returns the current platform's default platform specification.
 func Default() MatchComparer {
-	major, minor, build := windows.RtlGetNtVersionNumbers()
-	return matchComparer{
-		defaults: Ordered(DefaultSpec(), specs.Platform{
-			OS:           "linux",
-			Architecture: runtime.GOARCH,
-		}),
-		osVersionPrefix: fmt.Sprintf("%d.%d.%d", major, minor, build),
+	// Windows also supports linux via LCOW
+	linux := specs.Platform{
+		OS: "linux",
+		Architecture: "amd64",
 	}
```

Comment:
I'm not sure if this applies everywhere. I think the default matcher on Windows should remain to be "Windows" (and vice-versa for Linux), but it should be possible for consumers of the containerd code (like nerdctl) to easily construct a matcher for other platforms

I guess this would already be possible if `nerdctl` on Windows (if LCOW is available) to use `platform.Ordered()` or `platform.Any()` https://github.com/containerd/containerd/blob/a43703fcba5432751059575bbd89d578aed69101/platforms/compare.go#L110-L120

Perhaps a utility function that allows to combine multiple matchers, something like;
 
```go
func NewOrderedPlatformComparer(matchers ...Matcher) MatchComparer {
	return orderedPlatformComparer{
		matchers: matchers,
	}
}
```

Reply:
Interesting idea. Basically you would turn off LCOW for anybody that doesnt explicitly turn it on? So in theory we would need to do this in Docker, CRI, nerdctl, ctr, and anyone else that calls containerd directly

Reply:
Yes, so I'm not sure if LCOW will be available on every Windows set up (even if that's the case, I think the _default_ match should likely be best matching "windows" (native) image).

That said; containerd is meant to be used as component to build a "final" product/project, and from that perspective should be less opinionated (but flexible), so I think it would make sense to allow the project/code that uses it to construct an appropriate matcher for their purpose.

Reply:
I'm on-board even though I don't get a vote lol. LCOW is not possible for everyone to use and so pulling a LCOW layer of a multi-os image seems wasteful by default on Windows.

Reply:
>I guess this would already be possible if nerdctl on Windows (if LCOW is available) to use platform.Ordered() or platform.Any()

It is possible and why I found this issue with the matchers because it was using those functions which used the "default" matcher which doesn't match windows with the prefixes.

It makes sense to me to drop this and allow it to be passed through were appropriate. The tests now cover the scenario where a user might want to create their own LCOW matcher on Windows.

For why this is here, I carried this through from the current implementation (though it [never matched properly](https://github.com/containerd/containerd/blob/604c462d7bc59f967511c923e21c86014a4d0891/platforms/defaults_windows.go#L50-L52)): https://github.com/containerd/containerd/blob/604c462d7bc59f967511c923e21c86014a4d0891/platforms/defaults_windows.go#L84-L88

I also saw this on Darwin matcher: https://github.com/containerd/containerd/blob/604c462d7bc59f967511c923e21c86014a4d0891/platforms/defaults_darwin.go#L40-L43

Reply:
removed linux from the default matcher on windows